testing/src/main/java/org/oppia/android/testing/RecyclerViewScrollingActions.kt

```diff
@@ -0,0 +1,50 @@
+package org.oppia.android.testing
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.scrollTo
+import androidx.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition
+import androidx.test.espresso.contrib.RecyclerViewActions.scrollToHolder
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import org.hamcrest.BaseMatcher
+import javax.inject.Inject
+
+/**
+ * Provides different scrolling actions for [RecyclerView]s.
+ * This is needed because different testing scenarios potentially require different scrolling
+ * actions for the tests to be successful.
+ * Use the methods in this class for all scrolling actions for [RecyclerView]s in Roboelectric and
+ * Espresso test scenarios.
+ * See #2209 for more information.
+ * */
+class RecyclerViewScrollingActions @Inject constructor(
+  val testCoroutineDispatchers: TestCoroutineDispatchers
+) {
+
+  /**
+   * Performs scrolling action to particular RecyclerView at a required position
+   * Use this method for all testing scenarios when scrolling to a RecyclerView at a particular
+   * position
+   */
+  fun scrollToPosition(recyclerViewId: Int, position: Int) {
+    onView(withId(recyclerViewId)).perform(
+      actionOnItemAtPosition<RecyclerView.ViewHolder>(
+        position,
+        scrollTo()
+      )
+    )
+    testCoroutineDispatchers.runCurrent()
+  }
+
+  /**
+   * Performs scrolling action to [RecyclerView] of a particular ViewType
```

Comment:
```suggestion
   * Performs scrolling action to [RecyclerView] of a particular ViewType designed by a view holder via [baseMatcher].
```

Some additional context (may need wrapping). Try to keep this & other KDocs as proper paragraphs, too (e.g. don't line break unless it's for wrapping or you want to communicate separate ideas (in which case there should be a blank documentation line, e.g. just '*' between the paragraphs).

Ditto for both in above KDocs.

Reply:
Noted. Updated the change in commit.